Upgrade to Swagger-UI v2.1.8-M1

audionerd opened the pull request:
Update submodule and sync assets for [Swagger-UI v2.1.8-M1](https://github.com/swagger-api/swagger-ui/releases/tag/v2.1.8-M1).

Handlebars was upgraded to 2.0. `swagger.js` was apparently removed (?).

Working for me on a 2.0 docs project. Fixes a few small output bugs I noticed (missing tag descriptions, "Response Class" not showing status code).

audionerd commented:
Just realized this is missing `lib/marked.js`

hammerdr commented:
There's a slight change needed in the html/css, as well. I am using this branch and needed to wrap the render call in another div:
```
  = render 'swagger_ui/swagger_ui', discovery_url: '/docs/api-docs.json'
```

became

```
.swagger-section
  = render 'swagger_ui/swagger_ui', discovery_url: '/docs/api-docs.json'
```

Probably easier to just handle it in the render call

audionerd commented:
Yes, it looks like [app/views/_swagger_ui.html.erb](https://github.com/d4be4st/swagger-ui_rails/blob/a11774ed2841fcb5c51751a40958f0eff0a5ffb7/app/views/swagger_ui/_swagger_ui.html.erb) is out of date with https://github.com/swagger-api/swagger-ui/blob/master/dist/index.html

carlisia commented:
:+1:  for putting this in, it works fine with my swagger 2.0 document.

BjRo commented:
What's the current state of this? Would love to get this merged

@audionerd I've forked your project and fixed the issue with `marked.js`. I've also added `underscore-min.map` because it raised another js error. Please take a look [here](https://github.com/BjRo/swagger-ui_rails/tree/swagger_ui_update) and update your pull request if you consider this useful

McRip commented:
:+1:

audionerd commented:
@BjRo awesome, thanks for the marked update! Added.

I didn't see an error on `underscore-min.map` though. Do you know what causes it or how to reproduce that error?

@hammerdr It seems like Swagger UI expects the `body` of your docs page to have a class of `swagger-section`, so I'm just going to note that in the README.

Also updated the README to reflect the swagger-ui version (2.1.8-M1) that I pulled from.

brainopia commented:
I use this PR as well, I hope it will be merged upstream.

brainopia commented:
I've published this branch as `swagger-ui_rails_audionerd` in the meantime. (I've needed it for a gemspec dependency).

alexvodovoz commented:
+1 on getting this merged

wintondeshong commented:
+1

eprothro commented:
Is this repo still canonical, or has @audionerd master become what is being used? @d4be4st are you still maintaining?

sethherr commented:
Seconding @eprothro - @d4be4st are you still maintaining? When will this be merged?

reiz commented:
@d4be4st Any news on this?
